Sources/Plasma/FeatureLib/pfConsoleCore/pfConsoleEngine.cpp

```diff
@@ -540,40 +507,25 @@ bool    pfConsoleEngine::FindPartialCmd( char *line, bool findAgain, bool preser
 //  groups. numToSkip specifies how many matches to skip before returning one
 //  (so if numToSkip = 1, then this will return the second match found).
 
-bool    pfConsoleEngine::FindNestedPartialCmd( char *line, uint32_t numToSkip, bool preserveParams )
+ST::string pfConsoleEngine::FindNestedPartialCmd(const ST::string& line, uint32_t numToSkip, bool preserveParams)
 {
-    pfConsoleCmd        *cmd;
-
-    
     /// Somewhat easier than FindPartialCmd...
-    cmd = pfConsoleCmdGroup::GetBaseGroup()->FindNestedPartialCommand( line, &numToSkip );
+    pfConsoleCmd* cmd = pfConsoleCmdGroup::GetBaseGroup()->FindNestedPartialCommand(line, &numToSkip);
     if (cmd == nullptr)
-        return false;
+        return {};
 
     /// Recurse back up and get the group hierarchy
-    line[ 0 ] = 0;
-    IBuildCmdNameRecurse( cmd->GetParent(), line );
-    strcat( line, cmd->GetName() );
-    strcat( line, " " );
+    ST::string name = cmd->GetName() + ' ';
+    pfConsoleCmdGroup* group = cmd->GetParent();
+    while (group != nullptr && group != pfConsoleCmdGroup::GetBaseGroup()) {
+        name = group->GetName() + '.' + name;
+        group = group->GetParent();
+    }
```

Comment:
This should probably use `ST::string_stream` to avoid extra allocations.

Reply:
This isn't easily implemented with `ST::string_stream`, because that only allows appending to the end, but the string here is built from back to front. I suppose the strings could be collected into a `std::vector` and then joined at the end, to avoid at least some intermediate allocations.